test/Test/Toml/Gen.hs

```diff
@@ -332,7 +332,15 @@ Array [Double (-563397.0197456297),Double (-308866.62837749254),Double (-29555.3
 Nested array of AnyValue:
 Array [Array [Text "ACyz38VcLz0hxwdFkHTU6PYK8h8CeaiEpI2xAaiZTKBQ3zC1W717cZY35lk8EAK6pPw3WvwIdNktxIV2LrvFSpU8ee6zkXvpvePitW9aspAeeOCF9Q9ry20y7skFZ2qShi7CSx8888zWIqyc8iBkoLNvq4fONLtuUqSw2SlNee4hDIwrnx5O4RuHW1dQfJcnC34h9S0DlIGYP08qq6QHxO4E0HE74cNmiViGm3xpDC8Ro5D8Y6p0FLSN1ELq9Lwm",Text "HhNv0LKICdlKxN"],Array [Integer 986479839551009895,Integer 8636972066308796678,Integer (-3464941350081979804),Integer (-6560688879547055621),Integer (-4749037439349044738)],Array []]
 -}
-genArray :: MonadGen m => m (Value 'TArray)
+genArray :: (MonadGen m, GenBase m ~ Identity) => m (Value 'TArray)
 genArray = Gen.recursive Gen.choice
     [Gen.choice $ map genArrayFrom noneArrayList]
     [Array <$> Gen.list (Range.constant 0 5) genArray]
+
+-- filters
+
+invalidEscape :: Text -> Bool
+invalidEscape t
+    | t == Text.empty = True
+    | Text.last t == '\\' = False
+    | otherwise = True
```

Comment:
Can you please some documentation to this function and some examples of valid and invalid strings? And ideally with the link to TOML spec where it says that `"\"` cannot be a valid key. I'm afraid that this this solution solves the problem partially. Is the following key valid?

```toml
"\x"
```

I think that no, but it still can be genarated. I'm okay to accept this PR as a partial solution, but let's document some cases of what this function covers and what it doesn't cover :slightly_smiling_face:

Reply:
Look at `validText` in the last commit if it's what you asked